app/Http/Controllers/BoardsController.php

```diff
@@ -26,8 +27,13 @@ public function __construct()
      */
     public function index(Request $request)
     {
-        $boards = Board::with('user')->get();
+        $boards = new Board;
+        if ($request->exists('param')) {
+            $boards = $this->filter($boards, $request->param);
+        }
+        $boards = $boards->with('user')->orderBy('created_at', 'id')->paginate(20)->appends(request('page'));
```

Comment:
You can simplify the filtering if you use the when() method on the query. https://laravel.com/docs/5.7/queries#conditional-clauses

Reply:
Okay, I updated it.